biofefi/services/preprocessing.py

```diff
@@ -140,6 +170,18 @@ def run_preprocessing(
     X = data.iloc[:, :-1]
     y = data.iloc[:, -1]
 
+    columns_to_drop = find_non_numeric_columns(X)
+    if columns_to_drop:
+        X = X.drop(columns=columns_to_drop)
+
+    convert_y = find_non_numeric_columns(y)
```

Comment:
since your function has an error case where it throws and exception, you should do this in a `try/except` block